File: backend/src/main/java/cliquee/backend/model/EventParticipant.java
package cliquee.backend.model;

import cliquee.backend.model.embedded.EventParticipantId;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.MapsId;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import lombok.Data;

@Data
@Entity
@Table(name = "event_participants")
public class EventParticipant {

  @EmbeddedId
  private EventParticipantId id;

  @ManyToOne(fetch = FetchType.LAZY)
  @MapsId("event_id")
  @JoinColumn(name = "event_id")
  private Event event;

  @ManyToOne(fetch = FetchType.LAZY)
  @MapsId("user_id")
  @JoinColumn(name = "user_id")
  private User user;

  private String status;
  private LocalDateTime response_time = LocalDateTime.now();
}

File: flaskBackend/app/models/event_participant.py
from datetime import datetime
from ..extensions import db

class Event_Participant(db.Model):
    event_id = db.Column(db.Integer, db.ForeignKey('event.id'), nullable=False, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'), nullable=False, primary_key=True)
    created_at = db.Column(db.DateTime, default=datetime.now())

    def __repr__(self):
        return f'<Event_participant {self.user_id} in {self.event_id}>'